docs/dev.md

```diff
@@ -0,0 +1,129 @@
+# Development
+
+This page describes how to run the development environment on a machine with `docker compose`.
+
+???+ abstract "Requirements"
+
+    - [docker](https://docs.docker.com/get-docker/)
+    - [docker compose](https://docs.docker.com/compose/install/)
+
+    Running the services outside of docker is possible, but not supported in this context.
+
+## ./ov
+
+The `ov` file is the primary Open Vault command line script. This contains a number of pre-built commands to do basic operations
+
+### Usage
+
+`./ov COMMAND [args]`
+
+```bash title="./ov --help"
+COMMANDS:
+
+  b | build        build the docker images
+  backup | dump    create a backup of the database
+  c | cmd          run a compose command
+  d | dev          start a development server
+  deploy           run a ./deploy command
+  i | init         run initialization script
+  h | help         prints this help text
+  m | manage       run a wagtail manage.py command
+  restore | load   restore a database backup
+  s | shell        run a django shell command with the app context
+```
+
+### Commands
+
+#### `b` | `build`
+
+: Build the docker images locally.
+
+!!! abstract "build a single image"
+
+    Additional docker arguments can be passed to this command.
+
+    For example, to build only a single image:
+
+    ```bash title="Build frontend"
+    ./ov b front
+    ```
+
+#### `backup` | `dump`
+
+: Create a database dump file with the timestamp as the filename.
+
+#### `c` | `cmd`
+
+: Run a `docker compose` command with the base config files in place.
+
+#### `d` | `dev`
+
+: Run Development Environment
+
+: Run the development environment, with `docker compose`, and follow container logs.
+
+!!! abstract "Compose arguments"
+
+    Additional compose arguments can be passed. For example, to rebuild the containers before running:
+
+    ```bash
+    ./ov d --build
+    ```
+
+#### `deploy`
+
+: Shortcut for `./deploy [command]`.
+
+: See [Deploy](/deploy) for detailed usage.
+
+#### `docs`
+
+: Build and run the documentation server, with live change reloading.
+
+#### `i` | `init`
+
+: Initialize a development environment.
```

Comment:
running `./ov i`, the builds worked fine, but then on the npm install step I got this error:

```
 => [ov-wag production 1/2] RUN pip install "gunicorn>=20.1.0,<20.2.0"                                                                                  2.0s
 => [ov-wag production 2/2] COPY . .                                                                                                                    0.1s

unknown shorthand flag: 'i' in -it
```

Reply:
Hmmmm... I don't get that on my machine. I actually get a different error from a fresh clone:

```
=> CACHED [ov-front dev 4/5] COPY package.json .                                                                                                                            0.0s
 => CACHED [ov-front dev 5/5] RUN npm i                                                                                                                                      0.0s
 => CACHED [ov-front production 1/2] COPY ./ .                                                                                                                               0.0s
 => CACHED [ov-front production 2/2] RUN npm run build                                                                                                                       0.0s
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "GID" variable is not set. Defaulting to a blank string. 

> postinstall
> remix setup node

sh: remix: not found
npm ERR! code 127
npm ERR! path /var/app
npm ERR! command failed
npm ERR! command sh -c -- remix setup node

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-08-12T18_21_25_834Z-debug-0.log
```

This is resolved with the combination of [dockerfile changes in ov-frontend PR#6](https://github.com/WGBH-MLA/ov-frontend/pull/6) plus changing to `target: dev` in [ov-deploy PR#32](https://github.com/WGBH-MLA/ov-deploy/pull/32)

Reply:
Regarding your error, does the same thing happen when running the `init` commands directly?

Also, what versions of docker and compose?
``` bash
$ docker compose version
Docker Compose version v2.7.0
$ docker version
Version:          20.10.12
```

Reply:
docker: `Version:          20.10.11`
docker compose: `Docker Compose version v2.2.1`

my versions are a bit behind, that could be the problem

running the commands directly:
`./ov build` works fine
`./ov c run -it front npm install` fails with the error I mentioned

I'll see if I can update my docker versions, maybe its a change in parsing options

Reply:
now I get the same remix error as you after upgrading